src/api/typings/core.d.ts

```diff
@@ -0,0 +1,95 @@
+import { Balance, Claims } from '../../wallet'
+import { RPCResponse } from '../../rpc'
+import { Fixed8 } from '../../utils'
+import { Transaction, TransactionOutput } from '../../transactions'
+
+export type net = 'MainNet' | 'TestNet' | string;
+export type signingFunction = (unsigned: Transaction, publicKey: string) => Transaction
+
+export interface AssetAmounts {
+  GAS?: number
+  NEO?: number
+}
+
+export interface History {
+  address: string
+  history: PastTransaction[]
+  name: string
+  net: net
+}
+
+interface PastTransaction {
+  GAS: number
+  NEO: number
+  block_index: number
+  gas_sent: boolean
+  neo_sent: boolean
+  txid: string
+}
+
+export interface apiConfig {
+  net: net
+  address: string
```

Comment:
The introduction of `account` would mean this is optional. Yes, its getting messy. I can feel a refactor coming up.

Reply:
yes, this was actually what I was going to suggest next. Passing around this mutating object is getting a bit unwieldy. Was probably more convenient at the start, but now that it's starting to mature and settle a bit, might make sense to make the functions that use this object have specific inputs and outputs

Reply:
just updated to optional for the sake of this PR tho